workers/python/kitchen_sink.py

```diff
@@ -1,269 +1,195 @@
+from temporalio.api.common.v1 import Payload
+from temporalio.common import RawValue, RetryPolicy
 
-
-@dataclass
-class WorkflowParams:
-    action_set: Optional[ActionSet]
-    action_set_signal: str
-
-
-@dataclass
-class ActionSet:
-    actions: Sequence[Action]
-    concurrent: bool
-
-
-@dataclass
-class Action:
-    result: Optional[ResultAction]
-    error: Optional[ErrorAction]
-    continue_as_new: Optional[ContinueAsNewAction]
-    sleep: Optional[SleepAction]
-    query_handler: Optional[QueryHandlerAction]
-    signal: Optional[SignalAction]
-    execute_activity: Optional[ExecuteActivityAction]
-    execute_child_workflow: Optional[ExecuteChildWorkflowAction]
-    nested_action_set: Optional[ActionSet]
-
-
-@dataclass
-class ResultAction:
-    value: Any
-    run_id: bool
-
-
-@dataclass
-class ErrorAction:
-    message: str
-    details: Any
-    attempt: bool
-
-
-@dataclass
-class ContinueAsNewAction:
-    while_above_zero: int
-
-
-@dataclass
-class SleepAction:
-    millis: int
-
-
-@dataclass
-class QueryHandlerAction:
-    name: str
-
-
-@dataclass
-class SignalAction:
-    name: str
-
-
-@dataclass
-class ExecuteActivityAction:
-    name: str
-    task_queue: str
-    args: Optional[Sequence[Any]]
-    count: int
-    index_as_arg: bool
-    schedule_to_close_timeout_ms: int
-    start_to_close_timeout_ms: int
-    schedule_to_start_timeout_ms: int
-    cancel_after_ms: int
-    wait_for_cancellation: bool
-    heartbeat_timeout_ms: int
-    retry_max_attempts: int
-    non_retryable_error_types: Optional[Sequence[str]]
-
-
-@dataclass
-class ExecuteChildWorkflowAction:
-    name: Optional[str]
-    count: int
-    args: Optional[Sequence[Any]]
-    params: Optional[WorkflowParams]
-
-
-def workflow_query(arg: str) -> str:
-    return arg
+from protos.kitchen_sink_pb2 import (
+    Action,
+    ActionSet,
+    ActivityCancellationType,
+    DoActionsUpdate,
+    ExecuteActivityAction,
+    WorkflowInput,
+    WorkflowState,
+)
 
 
 @workflow.defn(name="kitchenSink")
 class KitchenSinkWorkflow:
-    @workflow.run
-    async def run(self, input: Optional[WorkflowParams]) -> Any:
-        workflow.logger.info("Started kitchen sink workflow {}".format(input))
-        if not input:
-            return None
-        if input.action_set and input.action_set.actions:
-            should_return, return_value = await self.handle_action_set(
-                input, input.action_set
-            )
-            if should_return:
-                return return_value
-
-        if input.action_set_signal:
-            action_set_queue: asyncio.Queue[ActionSet] = asyncio.Queue()
-
-            def enqueue(item: ActionSet) -> None:
-                action_set_queue.put_nowait(item)
+    action_set_queue: asyncio.Queue[ActionSet] = asyncio.Queue()
+    workflow_state = WorkflowState()
+
+    @workflow.signal
+    async def do_actions_signal(self, action_set: ActionSet) -> None:
+        self.action_set_queue.put_nowait(action_set)
```

Comment:
Didn't  we say we want to treat signal like update and allow for the action set to be run in the signal handler?

Reply:
Yes, that's right, I forgot to go back and add that. We talked about the DSL having an option to either pipe back to main to to run in-handler.